sdk/lib/rtl/rangelist.c

```diff
@@ -13,16 +13,478 @@
 #define NDEBUG
 #include <debug.h>
 
+/* GLOBALS ******************************************************************/
+
+//extern PAGED_LOOKASIDE_LIST RtlpRangeListEntryLookasideList;
+
 /* TYPES ********************************************************************/
 
-typedef struct _RTL_RANGE_ENTRY
-{
-    LIST_ENTRY Entry;
-    RTL_RANGE Range;
-} RTL_RANGE_ENTRY, *PRTL_RANGE_ENTRY;
+/* RTLP_RANGE_LIST_ENTRY == RTL_RANGE + ListEntry */
+
+/*  FIXME ? ntddk.h */
+/* Flags for public functions: RtlFindRange(), RtlIsRangeAvailable() */
+#define RTL_RANGE_LIST_SHARED_OK           1
+#define RTL_RANGE_LIST_NULL_CONFLICT_OK    2
+/* ? ntddk.h */
+
+/* RTLP_RANGE_LIST_ENTRY.PrivateFlags */
+#define RTLP_ENTRY_IS_MERGED  1
+
 
 /* FUNCTIONS ***************************************************************/
 
+static
+PRTLP_RANGE_LIST_ENTRY
+FASTCALL
+RtlpEntryFromLink(PVOID Link)
+{
+    return CONTAINING_RECORD(Link, RTLP_RANGE_LIST_ENTRY, ListEntry);
+}
+
+static
+BOOLEAN
+FASTCALL
```

Comment:
Please, remove the calling convention from all the private functions - let the compiler decide what's better here.
Also, it looks like all the functions except `NTSYSAPI` ones should be made `static`

Reply:
Ok